### PR TITLE
Docs: Improve Box3 page.

### LINKED ISSUE #19030: Bounding box of object containing concave geometry
##### Description of the problem

I have been trying to compute the bounding box of an object containing a concave geometry (with `new Box3().setFromObject()`) without success : the resulting bounding box is too big.

After a lot of digging in the three.js source code to understand how it works in details, I now fully understand why this happens : the bbox of the concave geometry is computed from local vertices, then the 8 corners of this bbox are transformed in world space (with `corner.applyMatrix4(object.matrixWorld)`) and finally the resulting bbox is the bbox from those transformed 8 corners in world space (with `Box3.setFromPoints(corners)`).

The issue is that with concave geometries, some corners of the local bbox are outside the geometry. It means that when we apply the world transform (typically containing a rotation), those points of the local bbox will have to be part of the final transformed bbox. This looks wrong to me, right ?

In the image below, we see in a 2D projection the corners of the local geometry bbox (in red), the final world bbox (in red) and the actual points and bbox I would personally expect from the computation (in orange, slightly shifted for you to see both bbox where they overlap in this particular case).

bbox

Of course now I know how to compute a correct bbox for this particular case :
`
new Box3().setFromPoints(geometry.vertices.map(v => v.clone().applyMatrix4(object.matrixWorld)))
`

But I find it quite cumbersome (since I cannot use `bbox.expandByObject(object)` without having to override the function) and it is very confusing (at least for new users) that computing a bounding box in the traditional way (the one we find it everywhere on three.js help forums) doesn't actually work as expected for concave geometries.

I know there is already some discussion about Geometry.boundingBox (#18334, for instance), but I am not sure this is the same issue. Is it a bug, a 'feature' (i.e. intended behavior) ? If this is the intended behavior, is there a better workaround than the one I wrote here ?

Thank you for considering.

##### Three.js version

r115

##### Browser

All of them

##### OS

All of them

##### Hardware Requirements (graphics card, VR Device, ...)

None in particular.

### PR DESCRIPTION
**Pull request by Mugen87**
Fixed #19030 by clarifying the documentation of AABB generation.

### CODE REVIEW

File: docs/api/en/math/Box3.html [hunk 1]
```diff
@@ -129,7 +129,7 @@ <h3>[method:Box3 expandByObject]( [param:Object3D object] )</h3>
 		Expands the boundaries of this box to include [page:Object3D object] and its children,
 		accounting for the object's, and children's, world transforms.
 		The function may result in a larger box than strictly necessary.
-
+		It's possible to get the precise box by using [page:.setFromBufferAttribute]() if desired.
```
Comment (WestLangley) [inline docs/api/en/math/Box3.html:132]: How do you account for the matrix transforms?

Comment (WestLangley) [reply docs/api/en/math/Box3.html:132]: >The function may result in a larger box than strictly necessary.

Instead, I would say:

    Note: For computational efficiency, the computed bounding box is conservative. That is, it is not necessarily the minimal bounding box.

Comment (Mugen87) [reply docs/api/en/math/Box3.html:132]: I'm afraid this note does not recommend what to do when you the exact bounding box. The was the actual problem of the OP in #19030.

> How do you account for the matrix transforms?

Well, you could clone the position attribute and then use `BufferAttribute.applyMatrix4()`.

Comment (WestLangley) [reply docs/api/en/math/Box3.html:132]: You are not using `Box3.setFromBufferAttribute()` as you proposed.

I would just omit the comment you are attempting to add here. Someone decided we are not going to return the minimal bounding box, so forget it.

Comment (Mugen87) [reply docs/api/en/math/Box3.html]: Well, it was the feature request of the OP. I let @mrdoob decide if the comment should be added. The new comment points user into the right direction.

Comment (WestLangley) [reply docs/api/en/math/Box3.html]: I am trying to help you.

As far as I can see, your comment is false.

Comment (Mugen87) [reply docs/api/en/math/Box3.html]: > As far as I can see, your comment is false.

Can you explain why?

Comment (WestLangley) [reply docs/api/en/math/Box3.html]: >It's possible to get the precise box by using [page:.setFromBufferAttribute]()

It requires doing something this:

```js
var attribute = attributes.position.clone();

attribute.applyMatrix4( object.matrixWorld );

box3.setFromBufferAttribute( attribute );
```
Getting it to work on a hierarchy of objects will require writing another method.

So the statement in the docs is not helpful. In fact, it is misleading. I would simply remove it.

Comment (Mugen87) [reply docs/api/en/math/Box3.html]: Don't you think users can figure out by themselves if it's necessary to honor the world matrix or the object hierarchy?  Mentioning that a solution needs use `setFromBufferAttribute()` or `expandByPoint()` is better than nothing from my point of view.

Comment (WestLangley) [reply docs/api/en/math/Box3.html]: The statement is misleading and certainly not sufficiently informative to be useful.

Look how much trouble I had trying to figure out what would be required -- and we have not even specified the entire program yet.

I would remove the statement.

Comment (Mugen87) [reply docs/api/en/math/Box3.html]: If you think it's not useful then I close the PR. We can still add something to the docs or the FAQ if more devs run into this issue. Feel free to change the existing text if you like. I think both are fine.

Comment (WestLangley) [reply docs/api/en/math/Box3.html]: @Mugen87 Thank you for considering my suggestion. :-)

I agree with you that we can leave it as-is for now.

**Comment by mrdoob** [comment]
https://github.com/mrdoob/three.js/pull/17940#issuecomment-611938083

**Pull request closed without merging**